src/test-run/index.js

```diff
@@ -106,6 +106,32 @@ export default class TestRun extends Session {
         this.injectable.scripts.push('/testcafe-automation.js');
         this.injectable.scripts.push('/testcafe-driver.js');
         this.injectable.styles.push('/testcafe-ui-styles.css');
+
+        this._initRequestHooks();
+    }
+
+    _initRequestHook (hook) {
+        hook.getRequestFilterRules().forEach(rule => {
+            this.requestFilterRuleHookMap.set(rule, hook);
+            this.addRequestFilterRule(rule);
+        });
+    }
+
+    _disposeRequestHook (hook) {
+        hook.getRequestFilterRules().forEach(rule => {
+            this.requestFilterRuleHookMap.delete(rule);
+            this.removeRequestFilterRule(rule);
+        });
+    }
+
+    _initRequestHooks () {
+        this.requestFilterRuleHookMap = new Map();
+
+        this.test.requestHooks.forEach(hook => this._initRequestHook(hook));
+
+        this.on(Session.REQUEST_EVENT, e => this.requestFilterRuleHookMap.get(e._requestFilterRule).onRequest(e));
```

Comment:
Move this logic to the Session. Add `Session.handleRequest(rule, { onRequest, onConfigure ,onResponse })`

Reply:
```js
Session.addRequestEventListeners (rule, { 
    onRequest: fn, 
    onConfigureRequest: fn,  
    onResponse: fn 
});

Session.removeRequestEventListeners (rule);
```

cc @AlexanderMoskovkin